homeassistant/components/apsystems/__init__.py

```diff
@@ -12,13 +14,22 @@
 
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
+ApsystemsConfigEntry = ConfigEntry["ApsystemsData"]
+
+
+@dataclass
+class ApsystemsData:
+    """Apsystems data type."""
+
+    coordinator: ApSystemsDataCoordinator
```

Comment:
You can also do `ConfigEntry[ApSystemsDataCoordinator]`, saves you the extra dataclass (maybe I pointed at the wrong integration for this)